ui/src/plugins/dev.perfetto.AndroidStartup/slowInsights.ts

```diff
@@ -0,0 +1,747 @@
+        AND slice_dur > ${threshold}
+    ) request
+    JOIN following_flow(request.id) arrow
+    JOIN slice reply ON reply.id = arrow.slice_in
+    JOIN thread USING (utid)
+    WHERE reply.dur > ${threshold} AND request.is_main_thread
+    ORDER BY request.slice_dur DESC
+    LIMIT 3`,
+    tag,
+    );
+    const it = result.iter({ts: LONG, dur: LONG, name: STR});
+    for (; it.valid(); it.next()) {
+      slices.push({
+        ts: it.ts,
+        dur: it.dur,
+        name: it.name,
+      });
+    }
+  }
+
+  const uri = `/android_startups/main_thread_binder_transactions`;
+  const title = `MainThreadBinderTransactions`;
+  return getSlowReasonTrack(trace, slices, uri, title);
+}
+
+async function getBroadcastDispatchedTrack(
+  trace: Trace,
+  startups: Startup[],
+): Promise<TrackNode | undefined> {
+  // The log tag
+  let tag = 'Broadcast dispatched';
+  let slice_glob = `'Broadcast dispatched*'`;
+
+  let slices = await getBroadcastSlices(trace, tag, slice_glob, startups);
+
+  const uri = `/android_startups/broadcast_dispatched`;
+  const title = `Broadcast Dispatched`;
+  return getSlowReasonTrack(trace, slices, uri, title);
+}
+
+async function getBroadcastReceivedTrack(
+  trace: Trace,
+  startups: Startup[],
+): Promise<TrackNode | undefined> {
+  // The log tag
+  let tag = 'Broadcast received';
+  let slice_glob = `'broadcastReceiveReg*'`;
+
+  let slices = await getBroadcastSlices(trace, tag, slice_glob, startups);
+
+  const uri = `/android_startups/broadcast_received`;
+  const title = `Broadcast Received`;
+  return getSlowReasonTrack(trace, slices, uri, title);
+}
+
+async function getBroadcastSlices(
+  trace: Trace,
+  tag: string,
+  slice_glob: string,
+  startups: Startup[],
+): Promise<Array<Slice>> {
+  const slices: Array<Slice> = [];
+
+  // The log tag
+  for (const startup of startups) {
+    let result = await trace.engine.query(
+    `
+    INCLUDE PERFETTO MODULE android.startup.startups;
+    SELECT s.ts AS ts, dur,  s.name AS name
+    FROM slice s
+    JOIN thread_track t ON s.track_id = t.id
+    JOIN thread USING(utid)
+    JOIN (
+      SELECT ts, ts_end
+      FROM android_startups
+      WHERE startup_id = ${startup.id}
+    ) launch
+    WHERE
+      s.name GLOB ${slice_glob} AND
+      s.ts BETWEEN launch.ts AND launch.ts_end
+    ORDER BY dur DESC
+    LIMIT 3`,
+    tag,
+    );
+    const it = result.iter({ts: LONG, dur: LONG, name: STR});
+    for (; it.valid(); it.next()) {
+      slices.push({
+        ts: it.ts,
+        dur: it.dur,
+        name: it.name,
+      });
+    }
+  }
+  return slices;
+}
+
+async function getOpenDexFilesFromOatTrack(
+  trace: Trace,
+  startups: Startup[],
+): Promise<TrackNode | undefined> {
+  // The log tag
+  let tag = 'OpenDexFilesFromOat';
+  let slice_glob = `'OpenDexFilesFromOat*'`;
+  let result = await trace.engine.query(
+    `
+    SELECT RUN_METRIC('android/startup/slow_start_thresholds.sql');
```

Comment:
Using RUN_METRIC in plugins is not allowed. There should be a presubmit check warning you about this if you did tools/install-build-deps. Please port all relevant things you need to the standard library first before trying to use them here.

Reply:
This applies here and for all other usages of RUN_METRIC in this class.